runtime.md

```diff
@@ -33,24 +33,79 @@ This is provided so that consumers can find the container's configuration and ro
 }
 ```
 
-## Lifecycle
+## Operations
+
+OCI compliant runtimes MUST support the following operations, unless support for the operation can not be supported by the base operating system (kernel) itself.
+
+OPEN: we should consider specifying the options/args for each operation.
+For example, checkpoint (for runc) seems to have a lot of params.
 
 ### Create
 
-Creates the container: file system, namespaces, cgroups, capabilities.
+Using the `config.json` and `runtime.json` files, along with the file system path referenced in the `config.json`, this operation MUST create a new container.
+This includes creating the relevant namespaces, cgroups and configuring the appropriate capabilities for the container.
+
+### Start
+
+This operation MUST start a previously created container.
+Starting a container MUST create a new process, as specified by the `config.json` file within the scope of the container.
+Attempting to start an already running container MUST have no effect on the container and MUST generate an error.
+Starting a stopped container, one that had been previously be running, MUST create a new process, as specified by the `config.json` file, within the scope of the container.
+
+OPEN: the entire "generate an error" stuff in here is something we should discuss.
+My initial thought was that it would be better to generate an error so that the call would know exactly what happened - meaning, did something new take place or was the container already running. If we didn't return an error (or something) then the caller wouldn't know.  And they can always ignore errors if they want/need to.
+
+### Stop
+
+This operation MUST stop a running container.
+Stopping a container MUST stop all of the processes running within the scope of the container.
+Stopping a container MUST NOT delete the associated namespaces or cgroups for the container.
+Attempting to stop a container that is not running MUST have no effect on the container and MUST generate an error.
+
+### Delete
+
+This operation MUST delete a container.
+Deleting a container MUST delete the associated namespaces and cgroups for the container.
+Deleting a container that is not stopped MUST first stop the container as specified by the `stop` operation.
+If the implied `stop` operations fails then the `delete` operation MUST fail.
+
+### Pause
+
+This operation MUST suspend all processes that are running within the container.
+If the container is not running, either stopped or aleady paused, then this operation MUST have no effect on the container and MUST generate an error.
+
+### Unpause
+
+This operation MUST resume all processes that were previously paused via the `pause` operation.
+If the container is not paused then this operation MUST have no effect on the container and MUST generate an error.
```

Comment:
On Wed, Oct 14, 2015 at 08:53:11PM -0700, Brandon Philips wrote:
> So, in order for this to work, a process that joins a PID namespace
> must join the freezer cgroup too?

I think the current “pause” description in this PR isn't clear enough
to decide whether it's going to work or not in a given situation ;).
If the definition is “pause all the processes in that container's
freezer cgroup”, then yes, it will work ;).  If you want to involve
PID namespaces somehow, then no, it probably won't work.  Personally,
I'm in favor of unified-cgroups, and a cgroup-based definition of
“container processes” [1], in which case the freezer approach will
work.

[1]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/n7vxlhn6Q7g